File: app/src/main/java/com/barikoi/cnlapp/RoomDb/AppDatabase.java
package com.barikoi.cnlapp.RoomDb;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.barikoi.cnlapp.order_create.RoomDB.DataConvertor;
import com.barikoi.cnlapp.order_create.RoomDB.OrderList;
import com.barikoi.cnlapp.order_create.RoomDB.OrderListDao;
import com.barikoi.cnlapp.order_create.RoomDB.SaveOrder;
import com.barikoi.cnlapp.order_create.RoomDB.SaveOrderDao;
import com.barikoi.cnlapp.Order_Delivery.RoomDB.UpdateOrder;
import com.barikoi.cnlapp.Order_Delivery.RoomDB.UpdateOrderDao;


@Database(entities = {OrderList.class, SaveOrder.class, UpdateOrder.class}, version = 6, exportSchema = false)
@TypeConverters(DataConvertor.class)
public abstract class AppDatabase extends RoomDatabase {

	public abstract OrderListDao orderListDao();
	public abstract SaveOrderDao saveOrderDao();
	public abstract UpdateOrderDao updateOrderDao();

	public static AppDatabase INSTANCE;
	private static final Object sLock = new Object();


	public static AppDatabase getInstance(Context context) {
		synchronized (sLock) {
			if (INSTANCE == null) {
				INSTANCE = Room.databaseBuilder(context,
						AppDatabase.class, "cnlapp.db")
						.allowMainThreadQueries()
						.fallbackToDestructiveMigration()
						.build();
			}
			return INSTANCE;
		}
	}
}

File: app/src/main/java/com/barikoi/cnlapp/imagecapture/RoomDb/AppDatabase.kt
package com.barikoi.cnlapp.imagecapture.RoomDb

import android.content.Context
import android.util.Log
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import java.util.concurrent.Executors


@Database(entities = [Images::class], version = 1, exportSchema = false)
abstract class AppDatabase() : RoomDatabase() {
    abstract fun imagesDao(): ImagesDao?

    companion object {
        var INSTANCE: AppDatabase? = null
        private val sLock = Any()
        private val NUMBER_OF_THREADS = 4
        val databaseWriteExecutor = Executors.newFixedThreadPool(
            NUMBER_OF_THREADS
        )
        /*val MIGRATION_1_2: Migration = object : Migration(1, 2) {
            override fun migrate(database: SupportSQLiteDatabase) {
                database.execSQL(
                    "ALTER TABLE VerifyImage "
                            + "ADD COLUMN Position INTEGER"
                )
            }
        }*/

        fun getInstance(context: Context): AppDatabase? {
            synchronized(sLock) {
                if (INSTANCE == null) {
                    Log.d("Room", "getInstance")
                    INSTANCE = Room.databaseBuilder(
                        context.getApplicationContext(),
                        AppDatabase::class.java, "ImageCapture.db"
                    )
                        .fallbackToDestructiveMigration()
                        .allowMainThreadQueries()
                        .build()
                }
                Log.d("Room", "getInstance: " + INSTANCE)
                return INSTANCE
            }
        }
    }
}